external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/EventHubSpout.java

```diff
@@ -17,246 +17,281 @@
+        String offset = messageId.getOffset();
+        partitionManager.ack(offset);
     }
 
-    if (eventDatawrap != null) {
-      MessageId messageId = eventDatawrap.getMessageId();
-      List<Object> tuples = scheme.deserialize(eventDatawrap.getEventData());
-      if (tuples != null) {
-        collector.emit(tuples, messageId);
-      }
+    @Override
+    public void fail(final Object msgId) {
+        MessageId messageId = (MessageId) msgId;
+        IPartitionManager partitionManager = partitionCoordinator.getPartitionManager(messageId.getPartitionId());
+        String offset = messageId.getOffset();
+        partitionManager.fail(offset);
     }
-    
-    checkpointIfNeeded();
-
-    // We don't need to sleep here because the IPartitionManager.receive() is
-    // a blocked call so it's fine to call this function in a tight loop.
-  }
-
-  @Override
-  public void ack(Object msgId) {
-    MessageId messageId = (MessageId) msgId;
-    IPartitionManager partitionManager = partitionCoordinator.getPartitionManager(messageId.getPartitionId());
-    String offset = messageId.getOffset();
-    partitionManager.ack(offset);
-  }
-
-  @Override
-  public void fail(Object msgId) {
-    MessageId messageId = (MessageId) msgId;
-    IPartitionManager partitionManager = partitionCoordinator.getPartitionManager(messageId.getPartitionId());
-    String offset = messageId.getOffset();
-    partitionManager.fail(offset);
-  }
-
-  @Override
-  public void deactivate() {
-    // let's checkpoint so that we can get the last checkpoint when restarting.
-    checkpoint();
-  }
-
-  @Override
-  public void close() {
-    for (IPartitionManager partitionManager : 
-      partitionCoordinator.getMyPartitionManagers()) {
-      partitionManager.close();
+
+    @Override
+    public void deactivate() {
+        // let's checkpoint so that we can get the last checkpoint when restarting.
+        checkpoint();
     }
-    stateStore.close();
-  }
-
-  @Override
-  public void declareOutputFields(OutputFieldsDeclarer declarer) {
-    if (Strings.isNullOrEmpty(eventHubConfig.getOutputStreamId())) {
-      declarer.declare(scheme.getOutputFields());
-    } else {
-      declarer.declareStream(eventHubConfig.getOutputStreamId(), scheme.getOutputFields());
+
+    @Override
+    public void close() {
+        for (IPartitionManager partitionManager : partitionCoordinator.getMyPartitionManagers()) {
+            partitionManager.close();
+        }
+        stateStore.close();
     }
-  }
 
-  private void checkpointIfNeeded() {
-    long nextCheckpointTime = lastCheckpointTime + checkpointIntervalInSeconds * 1000;
-    if (nextCheckpointTime < System.currentTimeMillis()) {
+    @Override
+    public void declareOutputFields(final OutputFieldsDeclarer declarer) {
+        List<String> fields = new LinkedList<String>();
+        fields.add(FieldConstants.MESSAGE_FIELD);
```

Comment:
You should use the fields declared by the scheme here, otherwise what's the point of having IEventDataScheme provide a declareOutputFields?